plugins/generator-1.20.1/forge-1.20.1/procedures/place_schematic.java.ftl

```diff
@@ -3,11 +3,11 @@ if (world instanceof ServerLevel _serverworld) {
 	StructureTemplate template = _serverworld.getStructureManager().getOrCreate(new ResourceLocation("${modid}", "${field$schematic}"));
 	if (template != null) {
 		template.placeInWorld(_serverworld,
-				${toBlockPos(input$x,input$y,input$z)},
-				${toBlockPos(input$x,input$y,input$z)},
-				new StructurePlaceSettings()
-						.setRotation(Rotation.${field$rotation!'NONE'})
-						.setMirror(Mirror.${field$mirror!'NONE'})
-						.setIgnoreEntities(false), _serverworld.random, 3);
+			${toBlockPos(input$x,input$y,input$z)},
+			${toBlockPos(input$x,input$y,input$z)},
+			new StructurePlaceSettings()
+				.setRotation(Rotation.<#if field$rotation?has_content && (field$rotation!'NONE')?string != "RANDOM">${field$rotation!'NONE'}<#else>values()[_serverworld.random.nextInt(3)]</#if>)

```

Comment:
Swap the check so you first check for random, then you can simplify this condition as right now this is quite messy FTL

Reply:
> Swap the check so you first check for random

Reply:
ok and then?

Reply:
> then you can simplify this condition

As in you don't need that many checks and even the cursed string cast

Reply:
Done